docs: Add ReadMe for Squish tests (#604)
aws-deadline/deadline-cloud: test/squish/SQUISH_README.md

# Squish GUI Submitter Tests

Squish requires a license. Currently, you may either have your own Squish license or you may file a [pull request](https://help.github.com/articles/creating-a-pull-request/) to the Deadline Cloud team to run or add any Squish automated tests against any changes to be committed. Please perform any necessary manual tests prior to submitting any changes, in addition to making sure at least a minimal render job test passes. If you have a Squish license, please follow our basic guide below to get set up. 

## Set Up/Install Deadline Cloud Client

If you haven't done so already, install Deadline Cloud with the GUI dependencies:
```sh
$ pip install "deadline[gui]"
```

> Verify that it was successfully installed by running `deadline config gui` and allow the Deadline Settings dialogue to load. 

Reviewer: ```suggestion
Verify that it was successfully installed by running `deadline config gui` and allow the Deadline Settings dialog to load. 
```

Not sure if it's specifically an American vs. British spelling but we use `dialog` everywhere to refer to the UI component so we should maintain consistency.

Author: Thx for these suggestions! Will update PR.

Author: It looks like I've been spelling it as 'dialogue', even in my tests (ie: `tst_verify_settings_dialogue`). I actually thought `dialog` was a mispelling. I do agree we should have some consistency, so I will do a follow-up PR to fix the spelling.

Reviewer: Ha, I will say dialogue looks right to me when talking about humans conversing

- AWS default profile
- A farm named "Deadline Cloud Squish Farm"
- A queue named "Squish Automation Queue"
- A storage profile named "Squish Storage Profile"

Note: The names of the above resources are in `./suite_deadline_gui/shared/config.py` file. 

Additionally, in config.py file, `simple_ui_with_ja` and `simple_ui_no_ja` file paths need to point to where those files live on your system when running `tst_verify_gui_submitter_bundles` tests locally.

## Running Squish Tests Locally

Squish tests can be run from the IDE or the command line. 
- In the IDE, hit the green 'Play' button located in the Test Suites left-hand column. 
- To run tests using the command line:
    - First, `cd ~/<squish_install_folder>/bin`
    - Example command to trigger all existing tests: 
    `./squishrunner --testsuite /home/<user>/deadline-cloud/test/squish/suite_deadline_gui --local`
    - Example command to trigger only `tst_verify_gui_submitter_bundles` test suite: 
    `./squishrunner --testsuite /home/<user>/deadline-cloud/test/squish/suite_deadline_gui --testcase tst_verify_gui_submitter_bundles --local`

## Adding Tests

Squish element locators:

> * Squish automatically identifies and creates all locators for you in a file called .`names.py`.

Reviewer: ```suggestion
* Squish automatically identifies and creates all locators for you in a file called `names.py`.
```

> * As the names of these locators do not necessarily have easily identifiable labels/names, all locators have been renamed and organized in it's own page. Please allow `names.py` file to be updated as new locators are identifed, and then copy/add new locators with easily identifiable labels/names into the appropriate files (ie: `workstation_config_locators.py`, `loginout_locators.py`, etc).

Reviewer: ```suggestion
* As the names of these locators do not necessarily have easily identifiable labels/names, all locators have been renamed and organized in their own page. Please allow `names.py` file to be updated as new locators are identified, and then copy/add new locators with easily identifiable labels/names into the appropriate files (ie: `workstation_config_locators.py`, `loginout_locators.py`, etc).
```

## A Final Word on Testing

> To ensure quality and reliability of any changes made to Deadline Client, please any necessary automated and manual tests prior to submitting changes.

Reviewer: ```suggestion
To ensure quality and reliability of any changes made to Deadline Client, please run any necessary automated and manual tests prior to submitting changes.
```
